### PR TITLE
test(ffi): smoke-test the C FFI boundary

### PR DESCRIPTION
**Pull request by louistrue**
Follow-up to #1182 (review point #4): the new `ifc-lite-ffi` crate was only *compiled* in CI, never exercised. This adds unit tests for the C ABI contract — important since these are `unsafe extern "C"` entry points.

## Coverage
- **null path / `out_ptr` / `out_len`** → code `1` (exercises the defensive null guards added in #1182)
- **invalid UTF-8 path** → code `1`
- **nonexistent file** → code `2`
- **well-formed minimal IFC** → code `0`, non-null buffer, valid JSON, then `ifc_lite_free`
- **`ifc_lite_parse_ex`** maps every `opening_filter_mode` (0/1/2 + out-of-range fallback)
- **`ifc_lite_free`** is a no-op on null / zero-len (no double-free / segfault)

## Notes
- Self-contained: embeds a minimal IFC4 string rather than reaching into another crate's fixtures, so the test owns its inputs.
- Unit tests live in `#[cfg(test)] mod tests` inside `lib.rs` — works for a `cdylib`-only crate without adding an `rlib` target.
- Test-only; no version bump / changeset needed.

```
running 6 tests
test result: ok. 6 passed; 0 failed
```

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive FFI boundary test coverage, including pointer handling validation, error code verification for edge cases, IFC4 parsing validation, and memory safety checks for buffer operations.

### CODE REVIEW

File: rust/ffi/src/lib.rs [hunk 1]
```diff
@@ -303,3 +303,149 @@ pub unsafe extern "C" fn ifc_lite_free(ptr: *mut u8, len: usize) {
         let _ = Box::from_raw(slice::from_raw_parts_mut(ptr, len));
     }
 }
+
+#[cfg(test)]
+mod tests {
```
[the remark this answers is not in the document]
Comment (louistrue) [reply rust/ffi/src/lib.rs]: Good catch — fixed in 6e72a90a. Moved the `#[cfg(test)]` module to a sibling `src/tests.rs` (declared `mod tests;`), bringing `lib.rs` back to 308 lines, under the ~400-line house rule. Kept it as a unit-test module rather than a `tests/` integration file because the crate is `cdylib`-only (no `rlib` target to link an integration test against). All 6 tests still pass.

**Comment by Sonderwoods** [comment]
wow youre fast. I was about to commit a smoke test but there you are already!

https://github.com/LTplus-AG/ifc-lite/commit/2ff8a991cedb482e91de3ed09d3bb95a17067bba (including symbolic data and ffi-smoke)
